### ISSUE TITLE
Admin Setting to adjust number of follows for Federated Timeline

### ISSUE THREAD

**Issue by ted-gould** [tootsuite/mastodon#7736]
I think that there should be an admin setting to adjust the number of local users that follow an external account before its toots show up in the federated timeline. The value exists today, but it is hardcoded to one. Instead I think that administrators should be able to set that value to adjust for the individual needs of their community.

What this effectively does is allows the community to vet external accounts as a group. So if multiple people follow that account it is likely that it isn't hostile to the local community. While it wouldn't remove all abuse, the other moderation tools that exist shouldn't be removed or other additions not considered, it does provide a way for external accounts to get moderated without involvement of admin. Which helps reduce the administration load of running an instance. It is also more likely to be more dynamic and responsive as if an external account starts with bad behavior while the administrators are unavailable, the users unfollowing would reduce the harm that account could cause on the federated timeline.

Shouldn't there be a universal value? After thinking about this some my conclusion is that there is no universal value for this setting because it effects how insular the community using the local instance is. Some communities would prefer to be mostly to themselves, with perhaps a few popular accounts being included, but others would want to be more open to anyone and everyone choosing to deal with bad actors as individual incidents. In this way one instance with 1,000 users could choose a value of 100 to be more local, while another could choose 5. Both values have merit depending on the local community's choices. An admin would know the local community's needs best.

Admins of popular instances may also like this setting to make the federated timeline feel like less of a firehose as well.

* * * *

- [X] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by nightpool** [comment]
Note that the value exists today only conceptually—the federated timeline actually shows **all** public statuses the instance knows about, of which the majority is because the author is followed by users on that instance, but it's not always the case (boosted posts, posts that are replied to a followed user, etc).

**Comment by ted-gould** [comment]
> all public statuses the instance knows about, of which the majority is because the author is followed by users on that instance

Yes, this makes the value effectively hardcoded to one. If anyone on the instance follows someone on another instance it shows up in the federated timeline. I think there is value in being able to set it higher.

**Comment by nightpool** [comment]
I agree, I'm just saying it's not as easy of a change as "The value exists today, but it is hardcoded to one" makes it sound.
